ray-curator/ray_curator/stages/audio/io/asr_inference.py

```diff
@@ -0,0 +1,143 @@
+from dataclasses import dataclass, field
+
+import nemo.collections.asr as nemo_asr
+
+from ray_curator.stages.base import CompositeStage, ProcessingStage
+from ray_curator.stages.io.reader.file_partitioning import FilePartitioningStage
+from ray_curator.tasks import SpeechObject, _EmptyTask
+from ray_curator.tasks.file_group import FileGroupTask
+
+
+@dataclass
+class AsrNemoInferenceStage(ProcessingStage[FileGroupTask, SpeechObject]):
+    """Stage that do speech recognition inference using NeMo model."""
+
+    model_name: str
+    filepath_key: str = "audio_filepath"
+    text_key: str = "text"
+    _name: str = "audio_inference"
+
+    def setup(self) -> None:
```

Comment:
Could you also implement `setup_on_node` to ensure that the model is already downloaded from Hugging Face? For example for text models we do: https://github.com/NVIDIA-NeMo/Curator/blob/bd176474788d1f5b3c3d3e2de15cfccae2294bdb/ray-curator/ray_curator/stages/classifiers/base.py#L95-L100